corehq/apps/translations/validator.py

```diff
@@ -128,9 +128,14 @@ def _map_ids_to_translations(self):
 
     @memoized
     def _get_current_header_index(self, module_or_form_id, header):
-        for index, _column_name in enumerate(self.current_headers[module_or_form_id]):
-            if _column_name == header:
-                return index
+        try:
+            for index, _column_name in enumerate(self.current_headers[module_or_form_id]):
+                if _column_name == header:
+                    return index
+        except KeyError:
+            raise BulkAppTranslationsException(_(
+                f"Could not find a module or form with ID='{module_or_form_id}' in app '{self.app.name}'"
+            ))
```

Comment:
nit: you could also just change the definition to
```
if module_or_form_id not in self.current_headers:
    raise BulkAppTranslationsException(_(
        f"Could not find a module or form with ID '{module_or_form_id}' in app '{self.app.name}'"
    ))
for index, _column_name in enumerate(self.current_headers[module_or_form_id]):
    if _column_name == header:
        return index
```
instead of going through the cycle of try except.

Don't think the "=" is needed.

Reply:
Is there any specific reason why `if` should be used instead of `try-catch`?

Reply:
There is no strong reason. The only think I can think of is to avoid the try-catch calls because we are already aware why the exception would happen. I generally use exception for unexpected scenarios. So I think it just comes to developer's preference. 
Feel free to punt this if you don't feel strongly about it, it's not a blocker.

Reply:
It makes sense that if we already know about the exception and can be handled by a simple check, then it should be done.
Update the code.